MATLAB R2015a crash while running quadrotor example using drake-distro compiled from sources

Hi,
I'm facing an issue where Matlab R2015a inexplicably crashes, while attempting to run the drake quadrotor example, using the latest version of the drake-distro (rigidbody branch) compiled from sources.

The current head commit id is
commit 26697d890939dc009fb1953c6e53f4e470845029
Author: Russ Tedrake
Date:   Wed Jun 24 06:40:29 2015 -0400

makefile works for non-cygwin windows

My platform configuration is Mac OS X 10.10.4, Xcode 6.4, MATLAB R2015a, MacPorts 2.3.3, with all required dependencies installed.

If I try to run the same example using the precompiled binary package (drake-distro-0.9.8 for Mac) with MATLAB R2015a, it works as expected.

I have attached the steps that I follow for the installation below:

cd /project/robotics/library/
git clone https://github.com/RobotLocomotion/drake-distro.git --recursive -b rigidbody

Install required dependencies:
$ sudo port install \
xorg \
xorg-libX11 +quartz\
fontconfig \
xft2 \
libiconv \
gmp \
gcc48 +gfortran\
tcl \
tk \
graphviz +tcl \
iverilog \
gtkwave


Set locale.

Export the following variables to set the locale:

export LC_ALL=en_US.UTF-8  
export LANG=en_US.UTF-8

Update the symlinks to use the GCC compiler installed by MacPorts

Type the following command to configure the system to use the MacPorts GCC compiler suite:

$ sudo port select --set gcc mp-gcc48

Check the binary pointed to by the c++ and g++ command:

$ which c++
/opt/local/bin/c++

$ which g++
/opt/local/bin/g++

By default, this doesn't automatically create a symlink for /opt/local/bin/cc, so you'll have to manually create the symlink to point to gcc for the c-compiler:

$ cd /opt/local/bin
$ sudo ln -s /opt/local/bin/gcc-mp-4.8 /opt/local/bin/cc

Check the binary pointed to by the cc command:

$ which cc
/opt/local/bin/cc

$ sudo port install wget gsed gtk2 freeglut

Note 01 : You might get the following error:
Package glu was not found in the pkg-config search path.
Perhaps you should add the directory containing `glu.pc'
to the PKG_CONFIG_PATH environment variable
Package 'glu', required by 'bot2-vis', not found
CMake Error at cmake/pods.cmake:300 (string):
  string sub-command STRIP requires two arguments.
Call Stack (most recent call first):
  src/wavefront-viewer/CMakeLists.txt:7 (pods_use_pkg_config_packages)

This is because freeglut needs to be installed as a dependency for bot-2-vis


Note 02: You might get the following error:


BUILD_PREFIX: /project/robotics/library/drake-distro/build


The Fortran compiler identification is unknown
CMake Error at CMakeLists.txt:4 (enable_language):
  No CMAKE_Fortran_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "FC" or the CMake cache entry CMAKE_Fortran_COMPILER to the full
  path to the compiler, or to the compiler name if it is in the PATH.


This is because even after install gcc48 using macports, you need to configure the system to use the MacPorts GCC suite.

Download and install Java for OS X 2014-001 for Mac OS X
https://support.apple.com/kb/DL1572?locale=en_US

Install pre-requisites

cd /project/robotics/library/drake-distro

sudo ./install_prereqs.sh macports

This will install gtk2 on Mac OS X 10.10.4, which will bring in a lot of packages and associated dependencies.

Build Drake and all dependent libraries:
make

Regards,

Elvis Dowson

--- a/systems/plants/viewer/drake_urdf_renderer.cpp
+++ b/systems/plants/viewer/drake_urdf_renderer.cpp
@@ -95,18 +95,21 @@ public:
     else if ( scale && num_scale_factors == 1 ) {
       scale_x = scale_y = scale_z = scale[0];
     }
+    cout << "Loading mesh: " << fname << endl;
     spruce::path path(fname);
     std::string ext = path.extension();
-    
+    cout << "Converting extension '" << ext << "'' to lowercase." << endl;
     std::transform(ext.begin(), ext.end(), ext.begin(), ::tolower);
     
     if (ext.compare(".obj")==0) {
-      pmesh = bot_wavefront_model_create(fname.c_str());
+      if (path.exists())
+        pmesh = bot_wavefront_model_create(path.getStr().c_str());
     } else  {
+      cerr << "Warning: Extension '" << ext << " was not expected.  Changing it to '.obj' and trying again." << endl;
+      // try changing the extension to obj and loading
       path.setExtension(".obj");
       if (path.exists())
-      // try changing the extension to obj and loading
-      pmesh = bot_wavefront_model_create(path.getStr().c_str());
+        pmesh = bot_wavefront_model_create(path.getStr().c_str());
     }
 
     if (!pmesh) {

